pkg/visitors/cat.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2018 the LinuxBoot Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package visitors
+
+import (
+	"regexp"
+
+	"github.com/linuxboot/fiano/pkg/uefi"
+)
+
+// Cat concatenates all RAW data sections from a file into a single byte slice.
+type Cat struct {
+	// Input
+	Predicate func(f *uefi.File, name string) bool
+
+	// Output
+	Buf     []byte
+	Matches []*uefi.File
+}
+
+// Run wraps Visit and performs some setup and teardown tasks.
+func (v *Cat) Run(f uefi.Firmware) error {
+	// First run "find" to generate a list of matches to replace.
+	find := Find{
+		Predicate: v.Predicate,
+	}
+	if err := find.Run(f); err != nil {
+		return err
+	}
+
+	v.Matches = find.Matches
+	for _, m := range v.Matches {
+		if err := m.Apply(v); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Visit applies the Extract visitor to any Firmware type.
+func (v *Cat) Visit(f uefi.Firmware) error {
+	switch f := f.(type) {
+
+	case *uefi.File:
+		return f.ApplyChildren(v)
+
+	case *uefi.Section:
+		if f.Header.Type == uefi.SectionTypeRaw {
+			v.Buf = append(v.Buf, f.Buf()[4:]...)
+		}
+		return f.ApplyChildren(v)
+
+	default:
+		// Must be applied to a File to have any effect.
+		return nil
+	}
+}
+
+func init() {
+	RegisterCLI("cat", 2, func(args []string) (uefi.Visitor, error) {
```

Comment:
The number `2` on this line is the number of arguments. It should probably be 1.

Reply:
gotcha.